lib/email_address.rb

```diff
@@ -13,59 +13,28 @@ module EmailAddress
     require "email_address/canonical_email_address_type"
   end
 
+  class << self
+    %i(valid? error normal redact munge canonical reference).each do |proxy_method|
+      define_method(proxy_method) do |*args, &block|
+        EmailAddress::Address.new(*args).send(proxy_method, &block)
+      end if EmailAddress::Address.method_defined? proxy_method
```

Comment:
Here are two more pretty options, in my opinion:

1.
```ruby
%i[some methods].each do |proxy_method|
  next unless EmailAddress::Address.public_method_defined? proxy_method
  # do logic
end
```

2.
```ruby
(%i[some methods] & EmailAddress::Address.public_instance_methods).each do |proxy_method|
  # do logic
end
```

Also, `method_defined?` is not very suitable because of:

> Public and protected methods are matched.

And `public_method_defined?` is more suitable.